table/tables/tables.go

```diff
@@ -72,12 +72,24 @@ func TableFromMeta(alloc autoid.Allocator, tblInfo *model.TableInfo) (table.Tabl
 		return nil, table.ErrTableStateCantNone.Gen("table %s can't be in none state", tblInfo.Name)
 	}
 
-	columns := make([]*table.Column, 0, len(tblInfo.Columns))
-	for _, colInfo := range tblInfo.Columns {
+	colsLen := len(tblInfo.Columns)
+	columns := make([]*table.Column, 0, colsLen)
+	for i, colInfo := range tblInfo.Columns {
 		if colInfo.State == model.StateNone {
 			return nil, table.ErrColumnStateCantNone.Gen("column %s can't be in none state", colInfo.Name)
 		}
 
+		// Add information about the column's offset.
+		if colInfo.Offset != i {
+			if i == colsLen-1 && colInfo.Offset >= colsLen {
+				log.Errorf("[tables] table %#v schema is wrong, col %#v, cols len %v", tblInfo, tblInfo.Columns[colsLen-1], colsLen)
```

Comment:
print colInfo.Offset?

Reply:
The offset is a field of `tblInfo.Columns[colsLen-1]`.